src/Controls/src/Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs

```diff
@@ -100,10 +100,7 @@ static bool IsPublicOrVisibleInternal(FieldDefinition fd, ModuleDefinition modul
 			{
 				if (fd.Module == module)
 					return true;
-				if (fd.Module.GetCustomAttributes().Any(ca => ca.AttributeType.FullName == "System.Runtime.CompilerServices.InternalsVisibleToAttribute"
-															&& ca.HasConstructorArguments
-															&& (ca.ConstructorArguments[0].Value as string) != null
-															&& (ca.ConstructorArguments[0].Value as string).StartsWith(module.Assembly.Name.Name, System.StringComparison.InvariantCulture)))
+				if (fd.Module.IsVisibleInternal(module))
```

Comment:
I extracted this logic out as I also wanted to use it - and it was already duplicated.